test-app/tests/integration/components/file-input-field-test.gts

```diff
@@ -0,0 +1,429 @@
+      />
+    </template>);
+
+    assert
+      .dom('[data-file-input-field]')
+      .hasAttribute('placeholder', 'Placeholder text');
+  });
+
+  test('it can accept a file using @onChange and display the filename', async function (assert) {
+    assert.expect(12);
+
+    class Context {
+      @tracked currentFiles: File[] = [] 
+    }
+
+    let ctx = new Context();
+   
+    // Note: typescript can't tell the assert is checking the event
+    // so it's complaining that the event is unused
+    /* eslint-disable @typescript-eslint/no-unused-vars */
+    const realOnChange = (files: File[], event: FileEvent) => {
+      const [firstFile] = files;
+
+      ctx.currentFiles = files; 
+
+      assert.ok(event, 'Expected `e` to be available as the only argument');
+      assert.ok(event.target, 'Expected direct access to target from `e`');
+      assert.ok(firstFile, 'Expected a single file to exist in the target');
+      assert.ok(firstFile instanceof File, 'Expected first file to be an instanceOf File');
+      assert.strictEqual(firstFile?.name, 'sample.txt', 'Expected the correct filename');
+      assert.strictEqual(firstFile?.size, 18, 'Expected the correct filename');
+      assert.strictEqual(files.length, 1, 'Expected a single file to be uploaded');
+      assert.step('realOnChange');
+    }
+
+    await render(<template>
+      <FileInputField 
+        @deleteLabel='Delete File'
+        @label="Label" 
+        @trigger="Select Files"
+        @onChange={{realOnChange}}
+        @files={{ctx.currentFiles}}
+        multiple="true"
+        data-file-input-field />
+    </template>);
+
+    assert.verifySteps([]);
+
+    const file = createFile(['Upload file sample'], {
+      name: 'sample.txt',
+      type: 'text/plain',
+    });
+
+
+    await triggerEvent('[data-file-input-field]', 'change', { files: [file] });
+
+    assert.verifySteps(['realOnChange']);
+
+    assert.dom('[data-files] [data-file-name]').hasText('sample.txt');
+
+    // currently: rounding to the nearest KB
+    // this file is very small so result is 0 KB
+    assert.dom('[data-files] [data-file-size]').hasText('0 KB');
+  });
+
+  test('it deletes a file', async function(assert) {
+    class Context {
+      @tracked currentFiles: File[] = [];
+    }
+
+    let ctx = new Context();
+
+    const realOnChange = (files: File[]) => {
+      ctx.currentFiles = files;
+    }
+
+    await render(<template>
+      <FileInputField
+        @deleteLabel='Delete File'
+        @label="Label"
+        @trigger="Select Files"
+        @onChange={{realOnChange}}
+        @files={{ctx.currentFiles}}
+        multiple="true"
+        data-file-input-field
+      />
+    </template>);
+
+    const file = createFile(['Upload file sample'], {
+      name: 'sample.txt',
+      type: 'text/plain',
+    });
+
+    await triggerEvent('[data-file-input-field]', 'change', { files: [file] });
+
+    await triggerEvent('button', 'click');
+
+    assert.dom('ul').doesNotExist();
```

Comment:
Nit: To ensure the file is added and then removed and to leverage some of the existing data attributes I think something like this would work:

```suggestion

    // Verify no files
    assert.dom('[data-files]').doesNotExist();
    
    await triggerEvent('[data-file-input-field]', 'change', { files: [file] });
    
    // Verify files are there
    assert.dom('[data-files]').exists();

    await triggerEvent('button', 'click');

    // Verify the `ul` is gone as all files are deleted
    assert.dom('[data-files]').doesNotExist();
```